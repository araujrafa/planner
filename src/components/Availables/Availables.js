import React from 'react';

export default ({ availables, selectCity, show }) => {
  const months = ['Janeiro', 'Fevereiro', 'Março', 'Abril', 'Maio', 'Junho', 'Julho','Agosto', 'Setembro', 'Outubro', 'Novembro', 'Dezembro'];

  const parseDate = (date) => {
    const splitDate = date.split('-');
    return [splitDate[2], months[+splitDate[1] - 1]];
  };

  const listResults = () => {
    const results = availables || [];
    // console.log(results)
    return results.map((elem, i) => {
      const date1 = parseDate(elem[0].date);
      const date2 = parseDate(elem[elem.length - 1].date);
      return (
        <li key={i}>
          De {date1[0]} de {date1[1]} à {date2[0]} de {date2[1]}
        </li>
      );
    });
  }

  return (
    <div className={`card ${show ? '' : 'd-none'}`} style={{marginTop: '20px'}}>
      <div className="card-body">
        <p>A saída mostra que, para {selectCity}, existem pelo menos {availables.length === 1 ? 'uma' : availables.length } combinações para o ano:</p>
        <ul>
          {availables.length > 0 ? listResults() : ''}
        </ul>      
      </div>
    </div>
    
  )
}