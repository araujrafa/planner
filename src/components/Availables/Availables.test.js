import React from 'react';
import Availables from './Availables';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';

describe('Availables component', () => {
  const props = {
    availables: [[{date: '2018-02-21'},{date: '2018-03-21'}]],
    show: true, 
    selectCity: 'SÃ£o Paulo'
  }

  test('Availables changes the class when hovered', () => {;
    const component = renderer.create(
      <Availables {...props}/>
    )
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
})