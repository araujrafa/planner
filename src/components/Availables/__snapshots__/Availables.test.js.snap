// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Availables component Availables changes the class when hovered 1`] = `
<div
  className="card "
  style={
    Object {
      "marginTop": "20px",
    }
  }
>
  <div
    className="card-body"
  >
    <p>
      A saída mostra que, para 
      São Paulo
      , existem pelo menos 
      uma
       combinações para o ano:
    </p>
    <ul>
      <li>
        De 
        21
         de 
        Fevereiro
         à 
        21
         de 
        Março
      </li>
    </ul>
  </div>
</div>
`;
